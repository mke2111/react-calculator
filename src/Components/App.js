/* eslint-disable no-unused-vars */
import React from 'react';
import DialPad from './DialPad';
import Result from './Result';
import '../index.css';
import calculate from '../Logic/Calculate';

function App() {
  return (
    <div className="container">
      <h1>Basic  Calculator</h1>
      <Result value="0" />
      <DialPad />
    </div>
  );
}

export default App;
