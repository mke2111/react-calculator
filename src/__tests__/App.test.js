import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import App from '../App';

describe('Calculator snapshot', () => {
  it('Renders the Calculator page', () => {
    const calculator = renderer
      .create(<App />)
      .toJSON();
    expect(calculator).toMatchSnapshot();
  });
});

describe('Calculator methods', () => {
  beforeEach(() => {
    render(<App />);
  });

  it('it test add numbers', async () => {
    userEvent.click(screen.getByText('6'));
    userEvent.click(screen.getByText('+'));
    userEvent.click(screen.getByText('6'));
    userEvent.click(screen.getByText('='));
    const display = await screen.findByText('12');
    expect(display).toBeInTheDocument();
  });

  it('test to subtract numbers', async () => {
    userEvent.click(screen.getByText('6'));
    userEvent.click(screen.getByText('8'));
    userEvent.click(screen.getByText('-'));
    userEvent.click(screen.getByText('6'));
    userEvent.click(screen.getByText('='));
    const display = await screen.findByText('62');
    expect(display).toBeInTheDocument();
  });

  it('test to multiply numbers', async () => {
    userEvent.click(screen.getByText('6'));
    userEvent.click(screen.getByText('8'));
    userEvent.click(screen.getByText('*'));
    userEvent.click(screen.getByText('2'));
    userEvent.click(screen.getByText('='));
    const display = await screen.findByText('136');
    expect(display).toBeInTheDocument();
  });

  it('test to divide numbers', async () => {
    userEvent.click(screen.getByText('6'));
    userEvent.click(screen.getByText('8'));
    userEvent.click(screen.getByText('/'));
    userEvent.click(screen.getByText('2'));
    userEvent.click(screen.getByText('='));
    const display = await screen.findByText('34');
    expect(display).toBeInTheDocument();
  });

  it('test for decimal numbers', async () => {
    userEvent.click(screen.getByText('6'));
    userEvent.click(screen.getByText('.'));
    userEvent.click(screen.getByText('2'));
    userEvent.click(screen.getByText('='));
    const display = await screen.findByText('6.2');
    expect(display).toBeInTheDocument();
  });
});
