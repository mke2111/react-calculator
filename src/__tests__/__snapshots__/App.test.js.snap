// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator snapshot Renders the Calculator page 1`] = `
<div
  className="container"
>
  <h1>
    Basic  Calculator
  </h1>
  <h2>
    0
  </h2>
  <div
    className="board-row"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      /
    </button>
  </div>
  <div
    className="board-row"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      *
    </button>
  </div>
  <div
    className="board-row"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="board-row"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="board-row"
  >
    <span>
      <button
        onClick={[Function]}
        type="button"
      >
        0
      </button>
    </span>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
